sdk/src/Services/DynamoDBv2/Custom/DataModel/ContextInternal.cs

```diff
@@ -359,6 +359,14 @@ private object FromDynamoDBEntry(SimplePropertyStorage propertyStorage, DynamoDB
             }
         }
         private bool TryFromList(Type targetType, DynamoDBList list, DynamoDBFlatConfig flatConfig, out object output)
+        {
+            var targetTypeWrapper = TypeFactory.GetTypeInfo(targetType);
+            return targetTypeWrapper.IsArray ?
+                 TryFromListToArray(targetType, list, flatConfig, out output) : //targetType is Array
+                 TryFromListToIList(targetType, list, flatConfig, out output) ; //targetType is IList or has Add method.
+        }
```

Comment:
It can reuse the `Utils.ItemsToCollection` method which is used by `PrimitiveCollectionConverter` and `DynamoDBListConverter` here.
If you would like this, I can include one more commit (https://github.com/jiabiao/aws-sdk-net/pull/1/commits/32543b36e750fa654ef9dbadb55c72c2fe64bb56) in this PR.